script.py

```diff
@@ -0,0 +1,152 @@
+import subprocess
+import os
+import boto3
+import argparse
+import logging
+from datetime import datetime, timedelta
+from dotenv import load_dotenv
+
+load_dotenv()
+logging.basicConfig(filename="app.log", level=logging.INFO)
+
+
+def mongodump(uri, outpath):
+    command = [
+        "mongodump",
+        "--uri",
+        uri,
+        "--out",
+        outpath,
+    ]
+    try:
+        subprocess.run(command, check=True)
+        logging.info("Backup completed successfully.")
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Backup failed: {e}")
+
+
+def uploadtos3(outpath, bucket, max_backups):
+    s3 = boto3.client("s3")
+    current_date = datetime.now().date()
+    path = None
+    datestr = None
+
+    try:
+        for i in range(1, max_backups + 1):
+            temp = s3.list_objects_v2(Bucket=bucket, Prefix=f"{current_date}_{i}")
+            if "Contents" in temp:
+                datestr = f"{current_date}_{i}"
+        if datestr:
+            no = int(datestr.split("_")[1])
+            if no == max_backups:
+                logging.error("Today's Backups are already stored. Come Back tomorrow")
+                return
+            path = f"{current_date}_{no + 1}"
+        else:
+            path = f"{current_date}_1"
+        for root, _, files in os.walk(outpath):
+            if (root == "admin") or (root == "config") or (root == "local"):
+                continue
+            for file in files:
+                local_path = os.path.join(root, file)
+                if file.endswith(".metadata.json"):
+                    key = (
+                        f"{path}/{os.path.split(root)[-1]}/{file.split('.')[0]}/{file}"
+                    )
+                else:
+                    key = f"{path}/{os.path.split(root)[-1]}/{os.path.splitext(file)[0]}/{file}"
+                s3.upload_file(local_path, bucket, key)
```

Comment:
add a callback by giving the output_file_path so we can get the progress percentage while file is being uploaded
`https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-uploading-files.html`